lib/composables/Url.ts

```diff
@@ -0,0 +1,98 @@
+import isPlainObject from 'lodash-es/isPlainObject'
+import { watch } from 'vue'
+import { useRoute, useRouter } from 'vue-router'
+
+export interface UseUrlQuerySyncOptions {
+  state: Record<string, any>
+  casts?: Record<string, (value: any) => any>
+  exclude?: string[]
+}
+
+export function useUrlQuerySync({
```

Comment:
Can we make the `state` to be an 1st arg? Since it's only required arg and if we don't need option, we can do:

```ts
useUrlQuerySync(options, { ... })
```